src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -70,6 +71,30 @@ public static class BitOperations
         [CLSCompliant(false)]
         public static bool IsPow2(ulong value) => (value & (value - 1)) == 0 && value != 0;
 
+        /// <summary>Round the given integral value up to a power of 2.</summary>
+        /// <param name="value">The value.</param>
+        internal static uint RoundUpToPowerOf2(uint value)
+        {
+            // TODO: https://github.com/dotnet/runtime/issues/43135
+            // When this is exposed publicly, decide on behavior be for the boundary cases...
```

Comment:
For reference, the main two concerns are `0` and `int.MaxValue + 2` through `uint.MaxValue`.

Given `0` is not a power of two, it makes sense for this to return `1`. This is what the `lzcnt` path returns, but the bithacks implementation returns `0` and needs an additional `value += (value == 0) ? 1 : 0`.

Likewise, the "logical" solution for `overflow` cases is to return `0`, representing the lowest 32-bits of a theoretically infinitely precise result. This is what the `bithacks` version returns (unless modified to account for rounding up) but the `lzcnt` variant ends up shifting by `32` which causes it to return `1` instead.

It would be great if we can figure out trivial additions to get the expected result in both cases without significant overhead.